### PR TITLE
FIX (GraphEngine): @W-13848149@: Adjusted violation URLs so they point to the proper anchors.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java [hunk 1]
```diff
@@ -15,6 +15,9 @@ public final class AvoidDatabaseOperationInLoop extends AbstractPathTraversalRul
     private static final ImmutableSet<ApexPathSource.Type> SOURCE_TYPES =
             ImmutableSet.copyOf(ApexPathSource.Type.values());
 
+    private static final String URL =
```
Comment (ganeshnt) [inline sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java:18]: do we have config system for string literals?

File: sfge/src/main/java/com/salesforce/rules/RemoveUnusedMethod.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public final class RemoveUnusedMethod extends AbstractPathBasedRule implements P
     private static final ImmutableSet<ApexPathSource.Type> SOURCE_TYPES =
             ImmutableSet.copyOf(ApexPathSource.Type.values());
     private static final String URL =
```
Comment (ganeshnt) [inline sfge/src/main/java/com/salesforce/rules/RemoveUnusedMethod.java:44]: same as above

File: sfge/src/main/java/com/salesforce/rules/UseWithSharingOnDatabaseOperation.java [hunk 3]
```diff
@@ -12,6 +12,9 @@
 
 public final class UseWithSharingOnDatabaseOperation extends AbstractPathTraversalRule {
 
+    private static final String URL =
```
Comment (ganeshnt) [inline sfge/src/main/java/com/salesforce/rules/UseWithSharingOnDatabaseOperation.java:15]: ditto

File: sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java (hunk 1 above)
Comment (ganeshnt) [reply sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java:18]: are there any integration tests that can hit these URLs in tests or test only URLs?

**Review by ganeshnt — APPROVED** [review]

File: sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java (hunk 1 above)
Comment (jfeingold35) [reply sfge/src/main/java/com/salesforce/rules/AvoidDatabaseOperationInLoop.java:18]: I don't think so. We could look into adding some in the future, though.

**Pull request merged**